pilot/pkg/xds/endpoint_builder.go

```diff
@@ -292,12 +292,13 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 	}
 	// The shards are updated independently, now need to filter and merge for this cluster
 	for _, clusterID := range keys {
-		endpoints := shards.Shards[clusterID]
 		// If the downstream service is configured as cluster-local, only include endpoints that
-		// reside in the same cluster.
-		if isClusterLocal && (cluster.ID(clusterID) != b.clusterID) {
+		// reside in the same cluster. A clusterID of "" may indicate this is a ServiceEntry registry and should be
+		// included.
+		if clusterID != "" && isClusterLocal && (cluster.ID(clusterID) != b.clusterID) {
```

Comment:
With this condition, all WLE can be visible even the service is cluster local.

Reply:
Yeah; won't merge as is. Planning on actually assigning a cluster-id.